charts/ceph-csi-cephfs/templates/nodeplugin-role.yaml

```diff
@@ -0,0 +1,21 @@
+{{- if .Values.rbac.create -}}
+{{- if and .Values.encryptionKMSConfig .Values.encryptionKMSConfig.secretNamespace  .Values.encryptionKMSConfig.secretName (eq .Values.encryptionKMSConfig.secretNamespace .Release.Namespace) -}}
+kind: Role
```

Comment:
Ceph-CSI components which reads the Secrets are usually running in a different namespace than the namespace where Secrets for applications have their encryption key. Ceph-CSI needs to read the Secret from a different namespace than where it is running/mounting the PV.

Reply:
In this case, it will create the `ClusterRole` automatically. The logic at line 2 is if `
`encryptionKMSConfig.secretNamespace` == `Release.Namespace`, then create a `Role` instead of adding `secret:read` to the `ClusterRole`.

Reply:
I think the optimization is nice to have, specially if users (like yourself) will benefit from it. Maybe document it, or explain it in a comment, and add it to the rbd chart too?

Reply:
No problem, I will put some comments in the README. I'll try to see if I can add it the the rbd too, is it okay if I do so in another PR tho? I seem to recall than `encryptionKMSConfig` doesn't work exactly the same and so it might be best to split the changes if that's okay with you?

Reply:
Thanks! Yes, a 2nd PR is fine for the RBD enhancement.

Reply:
I have slightly reworked the logic behind it to enforce determinism and added a some documentation in the `REAMDE.md`. I'd be nice if you could tell me how that looks, and if you like it, I can spin up a PR with the same feature for RBD.

Reply:
@nixpanic it'd be great if you could let me know your though on this

Reply:
Up to you! If it is ready to be merged, happy to do so too, and I'll wait and see if @nixpanic likes this approach for RBD and I can make the PR then

Reply:
@nixpanic friendly ping in case you missed the notification :)